scripts/regression/test_runner.py

```diff
@@ -151,32 +161,70 @@ class BenchmarkResult:
         ):
             regression_list.append(BenchmarkResult(benchmark, old_res, new_res))
         else:
-            other_results.append(BenchmarkResult(benchmark, old_res, new_res))
+            try:
+                improved_by = int((float(old_res) - float(new_res)) * 100.0 / float(old_res))
+            except Exception:
+                improved_by = 0
+            if improved_by > 0:
+                improvements_list.append(BenchmarkResult(benchmark, old_res, new_res, improvement=improved_by))
+            else:
+                other_results.append(BenchmarkResult(benchmark, old_res, new_res))
     benchmark_list = [res.benchmark for res in regression_list]
 
+time_old = geomean(old_runner.complete_timings)
+time_new = geomean(new_runner.complete_timings)
+
 exit_code = 0
 regression_list.extend(error_list)
 summary = []
 if len(regression_list) > 0:
-    exit_code = 1
-    print(
-        '''====================================================
-==============  REGRESSIONS DETECTED   =============
+    if not (isinstance(time_old, str) or isinstance(time_new, str)):
```

Comment:
can we also make sure `time_new > time_old`?
if `time_old > time_new` there are some regressions, but overall more improvements. Then you can just skip straight toward `regressions_header = 'ALLOWED_REGRESSIONS'`